public/models/HealGem.js

```diff
@@ -0,0 +1,38 @@
+if (typeof exports !== 'undefined' && typeof module !== 'undefined' && module.exports) {
+    // used on the server
+    // eslint-disable-next-line global-require
+    Gem = require('./Gem').Gem;
+}
+
+class HealGem extends Gem { // Dagmara
+    constructor(x, y, width, height, value, affectsMe) {
+      super(x, y, width, height, value, affectsMe);
+      this.values = [0.5, 1, 2.5, 3, 4.5];
+    }
+  
+    onCollect(player) {
+      this.healPlayer(player);
+    }
+  
+    //decide whether to heal current player or other players
+    healPlayer(player) {
+      if (this.affectsMe) {
+        this.heal(player);
+      } else {
+        socket.emit('healPlayer');
```

Comment:
I think you still need to add this to app.js

Reply:
you mean the socket?

Reply:
yes cause you emit it to server but you don't handle it on the server

Reply:
okok I see now